relayer/relayer/src/trader/mod.rs

```diff
@@ -0,0 +1,303 @@
+
+                    if let Err(why) = router
+                        .swap_exact_native_for_tokens(
+                            &azero_signed_connection,
+                            azero_available_for_swap,
+                            min_weth_amount_out,
+                            &swap_path,
+                            whoami_azero.clone(),
+                            now.saturating_add(HOUR_IN_MILLIS.into()) as u64, // within one hour
+                        )
+                        .await
+                    {
+                        warn!("Could not perform swap: {why:?}.");
+                        continue;
+                    }
+
+                    // check azero Eth balance
+                    let azero_eth_balance = match azero_ether
+                        .balance_of(azero_signed_connection.as_connection(), whoami_azero.clone())
+                        .await {
+                            Ok(balance) => balance,
+                            Err(why) => {
+                                warn!("Error when querying for Azero ETH balance: {why:?}.");
+                                continue;
+                            },
+                        };
+
+                    // Never bridge less than 0.1 ETH
+                    if azero_eth_balance > ONE_ETHER / 10 {
+                        info!("Requesting a cross chain transfer of {azero_eth_balance} units of Azero ETH [{azero_ether_address}] to {whoami_eth}.");
+
+                        // set allowance
+                        if let Err(why) = azero_ether.approve(&azero_signed_connection, most_azero.address.clone(), azero_eth_balance).await {
+                            warn!("Approve tx failed: {why:?}.");
+                            continue;
+                        }
+
+                        if let Err(why) = most_azero
+                            .send_request_native_ether(
+                                &azero_signed_connection,
+                                azero_eth_balance,
+                                receiver,
+                                current_base_fee
+                            )
+                            .await
+                        {
+                            warn!("Could not send the cross-chain transfer request: {why:?}.");
+                            continue;
+                        }
+                    }
+
+                    // check ETH balance
+                    if let Ok (eth_balance) = eth_signed_connection.get_balance(eth_signed_connection.address(), None).await {
+                        // warning if the balance drops too low
+                        if eth_balance < ONE_ETHER.into () {
+                            warn!("{whoami_eth} has a low ETH balance: {eth_balance} Wei.");
+                        } else {
+                            info!("{whoami_eth} has a balance of {eth_balance} Wei.");
+                        }
+                    }
```

Comment:
Question: would it be painful to extract smaller functions from, this, for example this piece, to keep the main one a manageable size? I know `async` can really make these kinds of refactors difficult, so might not be worth it...

Reply:
pita